packages/material-ui/src/utils/integerPropType.test.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { expect } from 'chai';
+import PropTypes from 'prop-types';
+import { getTypeByValue } from './integerPropType';
+import { integerPropType } from '.';
+
+describe('integerPropType', () => {
+  const location = '';
+  const componentName = 'DummyComponent';
+
+  function checkPropType(props, propName, required) {
+    PropTypes.checkPropTypes(
+      {
+        [propName]: required ? integerPropType.isRequired : integerPropType,
+      },
+      props,
+      '',
+      'DummyComponent',
+    );
+  }
+
+  function assertPasses({ props }, propName, required = false) {
+    expect(() => {
+      checkPropType(props, propName, required);
+    }).not.toErrorDev();
+  }
+
+  function assertFails({ props }, propName, required = false) {
+    const propType = getTypeByValue(props[propName]);
+    const errorMessage = `Warning: Failed  type: Invalid ${location} \`${propName}\` of type \`${propType}\` supplied to \`${componentName}\`, expected \`integer\`.`;
+
+    expect(() => {
+      checkPropType(props, propName, required);
+    }).toErrorDev(errorMessage);
+  }
+
+  describe('passes on undefined but failes on null value', () => {
+    beforeEach(() => {
+      PropTypes.resetWarningCache();
+    });
+
+    it('passes on undefined', () => {
+      assertPasses(<div a={undefined} />, 'a');
```

Comment:
We are always testing the prop `a`. It seems that we can abstract it.

Reply:
Please don't dry out tests needlessly.

Reply:
@oliviertassinari should I abstract these repetitions? If so, I could set the default value of propName to `a` and these repetitions would be deleted from the code.

Reply:
@fayzzzm No strong preferences. We are very cautious with abstractions in the tests (a different tradeoff compared to the code). We tend to only abstract when there is a strong duplication signal. For instance, the `fireEvent`, or `render` helpers have been abstracted because almost always identical. It also helps with the semantics.

Reply:
So, here, I think that the question we should ask ourselves is: Do we intertwine different test cases by abstracting? Do we see a future where the options have a slight chance to be used? If the answer is no to both questions. Then we could abstract.

I would personally make 'a' a default argument and remove it from all the cases, as we can always use this default value.

Reply:
Agree. For now, we don't need to point to a certain props field, as we have only one property. But, as said, in the future, it could be used to validate certain props fields' of components, so we have to provide the concrete `propName`.